### DISCUSSION TITLE
Installing @next/bundle-analyzer as dev dependency

### DISCUSSION THREAD

**Discussion by jacklimwenjie** [vercel/next.js#29697] [Help]
I am looking to install `@next/bundle-analyzer` as a dev dependency because I don't want it to be part of the production build.

There's a note on the README file:

> Note: if installing as a devDependency make sure to wrap the require in a process.env check as next.config.js is loaded during next start as well.

It seems to be a little vague to me. It would be good to have an example of setting up for dev dependency.

What I have in mind right now is to conditionally assign the `withBundleAnalyzer` variable with a pass-through function if process.env.ANALYZE is not `"true"`:

```
const withBundleAnalyzer = process.env.ANALYZE === "true" 
    ? require('@next/bundle-analyzer')() 
    : x => x; 
```

But I wonder is there a better way though? Or in fact there's little issue in installing it as a dependency instead of dev dependency?

**Comment by elbro** [comment]
yeah super confusing. I think its not included in prod build anyway though as a regular dep

**Comment by landsman** [comment]
I would love to install it as dev dependency as well. The suggested way is weird.
